src/OpenTelemetry.Contrib.Extensions.AWSXRay/Resources/AWSECSResourceDetector.cs

```diff
@@ -0,0 +1,90 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+#if NETSTANDARD
+using System;
+using System.Collections.Generic;
+
+namespace OpenTelemetry.Contrib.Extensions.AWSXRay.Resources
+{
+    /// <summary>
+    /// Resource detector for application running in AWS ECS.
+    /// </summary>
+    public class AWSECSResourceDetector : IResourceDetector
+    {
+        private const string AWSECSMetadataPath = "/proc/self/cgroup";
+        private const string AWSECSMetadataURLKey = "ECS_CONTAINER_METADATA_URI";
+        private const string AWSECSMetadataURLV4Key = "ECS_CONTAINER_METADATA_URI_V4";
+
+        /// <summary>
+        /// Detector the required and optional resource attributes from AWS ECS.
+        /// </summary>
+        /// <returns>List of key-value pairs of resource attributes.</returns>
+        public IEnumerable<KeyValuePair<string, object>> Detect()
+        {
+            List<KeyValuePair<string, object>> resourceAttributes = null;
+
+            if (!this.IsECSProcess())
+            {
+                return resourceAttributes;
+            }
+
+            try
+            {
+                var containerId = this.GetECSContainerId();
+
+                resourceAttributes = new List<KeyValuePair<string, object>>()
+                {
+                    new KeyValuePair<string, object>(AWSSemanticConventions.AttributeCloudProvider, "aws"),
+                    new KeyValuePair<string, object>(AWSSemanticConventions.AttributeCloudPlatform, "aws_ecs"),
+                    new KeyValuePair<string, object>(AWSSemanticConventions.AttributeContainerID, containerId),
+                };
+            }
+            catch (Exception ex)
+            {
+                AWSXRayEventSource.Log.ResourceAttributesExtractException(nameof(AWSECSResourceDetector), ex);
+            }
+
+            return resourceAttributes;
+        }
+
+        private string GetECSContainerId()
+        {
+            string containerId = null;
+
+            using (var streamReader = ResourceDetectorUtils.GetStreamReader(AWSECSMetadataPath))
+            {
+                while (!streamReader.EndOfStream)
+                {
+                    var trimmedLine = streamReader.ReadLine().Trim();
+                    if (trimmedLine.Length > 64)
+                    {
+                        containerId = trimmedLine.Substring(trimmedLine.Length - 64);
+                        return containerId;
+                    }
+                }
+            }
+
+            return containerId;
```

Comment:
We have a [try-catch](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/149/files/e9c374917b04fed78254d1cab9581bf2a4371246#diff-c9f4ccd7acaa7b04d4db40132e5890a872d9cc2124abd914333c10975a1eec91R56) block to deal with that so that we won't break user's application.